src/python/pants/core/goals/run.py

```diff
@@ -152,6 +177,9 @@ async def run(
 
         args = (arg.format(chroot=tmpdir) for arg in request.args)
         env = {**complete_env, **{k: v.format(chroot=tmpdir) for k, v in request.extra_env.items()}}
+        if run_subsystem.debug_adapter:
+            logger.info("Launching debug adapter, which will wait for a client connection...")
```

Comment:
I'd like to print the host/port, but right now that is isolated to `debugpy`. I think we should add a new subsystem for `[debug-adatper]` and centralize generic options there.